### PR TITLE
Prettier precommit hook

### LINKED ISSUE #134: Prettier pre-commit hook
### ⚙️ Improvement
Leverage [Husky](https://github.com/typicode/husky) and [Precise-Commits](https://github.com/nrwl/precise-commits) with a [pre-commit hook](https://github.com/nrwl/precise-commits#2-precommit-hook) in order to format per commit, ensuring that linting and formatting is constantly maintained while avoiding any git pollution.

### ❓Reasoning
I have Prettier integrated into my editor, and your codebase uses Prettier; however, files required formatting before making any changes. This indicates one or both of the following are true:
1) The entire codebase needs formatting against using the command:

```prettier --write "**/*.(js|less)"```

2) Incorrectly formatted code is entering the code base.

---

This improvement prevents number 2 from ever happening again, and number 1 can be solved permanently by running the aforementioned command _after_ this issue is resolved. 😎 

(I'd love to take this if it's something y'all desire)

### PR DESCRIPTION
**Pull request by kylemh**
**Link to issue:**
Resolves #134 

**Short description:**

Every commit will have it's staged lines prettified now 💅 

**PR Checklist:**

* [ ] Added tests for bugs / new features
* [ ] Updated docs (storybooks, readme)
* [X] Described changes in CHANGELOG.md
* [X] Executed `npm run check` and made sure no errors / warnings were shown
* [X] Executed `npm run test` and made sure all tests are passing
* [X] Bumped version in package.json
* [ ] Updated all static build artifacts (`npm run build-all`)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Reactist follows [semantic versioning](https://semver.org/) and doesn't introduce breaking changes (API-wise) in minor or patch releases. However, the appearance of a component might change in a minor or patch release so keep an eye on redesigns and make sure your app still looks and feels like you expect it.
 
+## 1.17.02
+
+-   [Tweak] Added pre-commit hook for formatting/linting
+
```
Comment (kylemh) [inline CHANGELOG.md]: the rest of the markdown file will get prettified in the next PR

**Comment by henningmu** [comment]
Awesome stuff. I expected more config for this work. I'm impressed and thanks for the contribution.

Two smaller things / questions:
1. What do you think about changing our `npm run check` command to run on all files instead of only js / jsx? I think that's what caused the undetected "errors" slipping in the codebase. Maybe `npm run check` could also just be an alias for `precise-commits`?
2. Do we really need to release a new version for this? It shouldn't affect the published code, right?

**Comment by kylemh** [comment]
I saw the `check` command, but also noticed you have no visible CI process.

If you're counting on people to run `npm run check` manually perhaps that simply isn't happening.

I don't think it justifies a new release version, was just following the checklist and some repos I've contributed to want a new version per merged branch.

I would be willing to help setup a TravisCI system which deploys your Storybook instance per pull request merge to master, while ensuring that linting and testing are compliant.

You can check out all the checks I have setup on [Operation Code's front-end](https://github.com/OperationCode/front-end/pull/276) and we could emulate something similar.

**Comment by henningmu** [comment]
Since I'm more or less the only contributor up until now `npm run check` has been run all the times. That's why I'm so shocked that the codebase got "unprettified" 😱 But I get your point that we don't need the check command when we have a prettified codebase and avoid regressions with the hook 👍

One thing left to be done:
```
Warning: Setting pre-commit script in package.json > scripts will be deprecated
Please move it to husky.hooks in package.json, a .huskyrc file, or a husky.config.js file
Or run ./node_modules/.bin/husky-upgrade for automatic update
```

Let's fix that warning and I'm good with the merge :)

**Comment by kylemh** [comment]
Will do!

**Comment by gnarmedia** [comment]
Wow, after a brief read of precise-commits it looks like it solves all my issues with lint-staged quite elegantly, I might start using this on my own projects and here at my current job :)

**Comment by gnapse** [comment]
@kylemh this would be a great addition. The only thing left is making sure this adheres to husky's newer way to specify the git hooks, as requested by @henningmu above. Are you still up to getting this done? Let me know if you need help, or if you prefer for us to finish it. Thanks!

**Comment by kylemh** [comment]
Sorry, I won't have time to commit to this project.

**Comment by gnapse** [comment]
Ok, thanks anyway for your interest.

I'm closing this as it's superseded by #162.

**Comment by gnarmedia** [comment]
I can look at fixing this up, I am curious how precise commit works so maybe I can work on this if you like.

**Comment by gnapse** [comment]
I already rolled out a new PR #162, and although I too was intrigued by `precise-commits` (I did not know it existed and it looks really good) I was interested in not just applying `prettier`, but a bunch of other stuff not only on commit, but also on push. So I sticked to `lint-staged`. You're more than welcome to checkout #162 and let me know what you think.

**Comment by kylemh** [comment]
FWIW `lint-staged` resolved their issue, and I'd use them over `precise-commits` nowadays as well.

When the PR was first opened, `lint-staged` would format and commit changes from an entire file, not just staged lines.

**Comment by gnapse** [comment]
> When the PR was first opened, lint-staged would format and commit changes from an entire file, not just staged lines.

Yes, I recall those days. Thanks for the info.

**Comment by gnarmedia** [comment]
Oh, cool, so he solved it! that's awesome, I recall that giant thread he mentions in his blog post...

**Pull request closed without merging**